### ISSUE TITLE
Proposal: Expose method to write error to http.ResponseWriter

### ISSUE THREAD

**Issue by ofpiyush** [twitchtv/twirp#161]
We need to check for some headers in a HTTP middleware and would love to return twirp style errors.

Right now we've copied code from generated code. That may or may not change between twirp spec versions and end up breaking our code

```go
type twerrJSON struct {
	Code string            `json:"code"`
	Msg  string            `json:"msg"`
	Meta map[string]string `json:"meta,omitempty"`
}

func Middleware(next http.Handler) http.Handler {
	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
		myVar := r.Header.Get("MY_KEY")
		if myVar == "" {
			err := twirp.InternalError("Could not find MY_KEY in request")
			w.Header().Set("Content-Type", "application/json")
			statusCode := twirp.ServerHTTPStatusFromErrorCode(err.Code())
			w.WriteHeader(statusCode)
			twirpErr := &twerrJSON{
				Code: string(err.Code()),
				Msg:  err.Msg(),
				Meta: err.MetaMap(),
			}
			json.NewEncoder(w).Encode(twirpErr)
			return
		}
		// More code here
	})
}
```

If the supported method exists, it can be as simple as
```go
func Middleware(next http.Handler) http.Handler {
	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
		myVar := r.Header.Get("MY_KEY")
		if myVar == "" {
			err := twirp.InternalError("Could not find MY_KEY in request")
			twirp.RespondWithError(w, err)
			return
		}
		// More code here
	})
}
```

**Comment by spenczar** [comment]
Yeah, this has been a fairly frequent source of pain.

One reason we have avoided doing this is that we fiddled with the encoding of errors in early (pre-public) versions of Twirp. By keeping it in the generated code, it was easier to make changes that broke compatibility, since all we needed was for the generated client and server to agree on an encoding, not anyone else.

As Twirp has matured into having a real specification for error encoding, that argument is less persuasive; we can't just break error encoding. I think this is probably worth doing. It would improve a lot of code.
